daml-foundations/daml-ghc/src/DA/Test/GHC.hs

```diff
@@ -175,7 +177,7 @@ runJqQuery log qs = do
     log $ "running jq query: " ++ q
 
     let jq = "external" </> "jq" </> "bin" </> "jq"
-    let json = unpack $ A.encode $ transformOn A._Value numToString $ A.toJSON pkg
+    let json = unpack $ A.encode $ transformOn A._Value numToString $ JSONPB.toJSONPB (encodePackage pkg) JSONPB.jsonPBOptions
```

Comment:
@martin-drhu-da Do you know why we need the `transformOn`? I was just finishing/testing my work on this and it looks like it's not needed. Also I used `encodePayload pkg` (from https://github.com/digital-asset/daml/blob/master/compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/Encode.hs#L12) instead of the `JSONPB` stuff. Looks like our results are the same though. :)

Reply:
IIRC, the `transformOn` encodes numbers as strings. This is a workaround for the circumstance that JSON numbers are all double and we sometimes lose precision.